### ISSUE TITLE
"PHP Deprecated" warnings with MW 1.36.0

### ISSUE THREAD

**Issue by webracer999** [SemanticMediaWiki/SemanticMediaWiki#5001]
### Setup

- SMW version: 3.2.3
- MW version: 1.36.0
- PHP version: 7.4.3 (fpm-fcgi)
- DB system (MySQL, Blazegraph, etc.) and version: MySQL 8.0.25-0ubuntu0.20.04.1

### Issue

With update to MW 1.36.0 I get numerous "PHP Deprecated" warnings.  
I also tested with composer `"mediawiki/semantic-media-wiki": "dev-master"` (version "4.0.0-alpha (487a909)", reported in MediaWiki), which results in the same warnings.

`php /var/www/wiki/extensions/SemanticMediaWiki/maintenance/rebuildData.php --quiet --shallow-update` or  
`php /var/www/wiki/extensions/SemanticMediaWiki/maintenance/rebuildData.php --quiet -d 100`:

```
PHP Deprecated:  Premature access to service container [Called from Hooks::run in /var/www/wiki/includes/Hooks.php at line 135] in /var/www/wiki/includes/debug/MWDebug.php on line 376
PHP Deprecated:  Premature access to service 'HookContainer' [Called from MediaWiki\MediaWikiServices::getInstance in /var/www/wiki/includes/MediaWikiServices.php at line 252] in /var/www/wiki/includes/debug/MWDebug.php on line 376
PHP Deprecated:  Premature access to service 'ObjectFactory' [Called from Wikimedia\Services\ServiceContainer::{closure} in /var/www/wiki/includes/ServiceWiring.php at line 535] in /var/www/wiki/includes/debug/MWDebug.php on line 376
PHP Deprecated:  Premature access to service 'StatsdDataFactory' [Called from ObjectCache::newFromParams in /var/www/wiki/includes/objectcache/ObjectCache.php at line 149] in /var/www/wiki/includes/debug/MWDebug.php on line 376
PHP Deprecated:  Premature access to service 'MainConfig' [Called from Wikimedia\Services\ServiceContainer::{closure} in /var/www/wiki/includes/ServiceWiring.php at line 1424] in /var/www/wiki/includes/debug/MWDebug.php on line 376
PHP Deprecated:  Premature access to service 'ConfigFactory' [Called from Wikimedia\Services\ServiceContainer::{closure} in /var/www/wiki/includes/ServiceWiring.php at line 734] in /var/www/wiki/includes/debug/MWDebug.php on line 376
PHP Deprecated:  Premature access to service 'BootstrapConfig' [Called from Wikimedia\Services\ServiceContainer::{closure} in /var/www/wiki/includes/ServiceWiring.php at line 277] in /var/www/wiki/includes/debug/MWDebug.php on line 376
PHP Deprecated:  Premature access to service 'DBLoadBalancer' [Called from ObjectCache::{closure} in /var/www/wiki/includes/objectcache/ObjectCache.php at line 181] in /var/www/wiki/includes/debug/MWDebug.php on line 376
PHP Deprecated:  Premature access to service 'DBLoadBalancerFactory' [Called from Wikimedia\Services\ServiceContainer::{closure} in /var/www/wiki/includes/ServiceWiring.php at line 383] in /var/www/wiki/includes/debug/MWDebug.php on line 376
PHP Deprecated:  Premature access to service 'MainWANObjectCache' [Called from Wikimedia\Services\ServiceContainer::{closure} in /var/www/wiki/includes/ServiceWiring.php at line 406] in /var/www/wiki/includes/debug/MWDebug.php on line 376
PHP Deprecated:  Premature access to service 'LocalServerObjectCache' [Called from Wikimedia\Services\ServiceContainer::{closure} in /var/www/wiki/includes/ServiceWiring.php at line 411] in /var/www/wiki/includes/debug/MWDebug.php on line 376
PHP Deprecated:  Premature access to service 'ConfiguredReadOnlyMode' [Called from Wikimedia\Services\ServiceContainer::{closure} in /var/www/wiki/includes/ServiceWiring.php at line 420] in /var/www/wiki/includes/debug/MWDebug.php on line 376
PHP Deprecated:  Use of Revision::__construct was deprecated in MediaWiki 1.31. [Called from SMW\MediaWiki\RevisionGuard::newRevisionFromTitle in /var/www/wiki/extensions/SemanticMediaWiki/src/MediaWiki/RevisionGuard.php at line 152] in /var/www/wiki/includes/debug/MWDebug.php on line 376
PHP Deprecated:  Use of Revision::getContent was deprecated in MediaWiki 1.31. [Called from SMW\ContentParser::fetchFromContent in /var/www/wiki/extensions/SemanticMediaWiki/includes/ContentParser.php at line 149] in /var/www/wiki/includes/debug/MWDebug.php on line 376
PHP Deprecated:  Use of Revision::getId was deprecated in MediaWiki 1.31. [Called from SMW\ContentParser::fetchFromContent in /var/www/wiki/extensions/SemanticMediaWiki/includes/ContentParser.php at line 159] in /var/www/wiki/includes/debug/MWDebug.php on line 376
PHP Deprecated:  Use of Revision::getUser was deprecated in MediaWiki 1.31. [Called from SMW\DataUpdater::runUpdate in /var/www/wiki/extensions/SemanticMediaWiki/src/DataUpdater.php at line 281] in /var/www/wiki/includes/debug/MWDebug.php on line 376
```

`php /var/www/wiki/extensions/SemanticMediaWiki/maintenance/rebuildConceptCache.php --quiet --update --create`,  
`php /var/www/wiki/extensions/SemanticMediaWiki/maintenance/rebuildPropertyStatistics.php --quiet`,  
`php /var/www/wiki/extensions/SemanticMediaWiki/maintenance/disposeOutdatedEntities.php --quiet`,  
`php /var/www/wiki/extensions/SemanticMediaWiki/maintenance/setupStore.php --quiet --skip-import`:

```


PHP Deprecated:  Premature access to service container [Called from Hooks::run in /var/www/wiki/includes/Hooks.php at line 135] in /var/www/wiki/includes/debug/MWDebug.php on line 376
PHP Deprecated:  Premature access to service 'HookContainer' [Called from MediaWiki\MediaWikiServices::getInstance in /var/www/wiki/includes/MediaWikiServices.php at line 252] in /var/www/wiki/includes/debug/MWDebug.php on line 376
PHP Deprecated:  Premature access to service 'ObjectFactory' [Called from Wikimedia\Services\ServiceContainer::{closure} in /var/www/wiki/includes/ServiceWiring.php at line 535] in /var/www/wiki/includes/debug/MWDebug.php on line 376
PHP Deprecated:  Premature access to service 'StatsdDataFactory' [Called from ObjectCache::newFromParams in /var/www/wiki/includes/objectcache/ObjectCache.php at line 149] in /var/www/wiki/includes/debug/MWDebug.php on line 376
PHP Deprecated:  Premature access to service 'MainConfig' [Called from Wikimedia\Services\ServiceContainer::{closure} in /var/www/wiki/includes/ServiceWiring.php at line 1424] in /var/www/wiki/includes/debug/MWDebug.php on line 376
PHP Deprecated:  Premature access to service 'ConfigFactory' [Called from Wikimedia\Services\ServiceContainer::{closure} in /var/www/wiki/includes/ServiceWiring.php at line 734] in /var/www/wiki/includes/debug/MWDebug.php on line 376
PHP Deprecated:  Premature access to service 'BootstrapConfig' [Called from Wikimedia\Services\ServiceContainer::{closure} in /var/www/wiki/includes/ServiceWiring.php at line 277] in /var/www/wiki/includes/debug/MWDebug.php on line 376
PHP Deprecated:  Premature access to service 'DBLoadBalancer' [Called from ObjectCache::{closure} in /var/www/wiki/includes/objectcache/ObjectCache.php at line 181] in /var/www/wiki/includes/debug/MWDebug.php on line 376
PHP Deprecated:  Premature access to service 'DBLoadBalancerFactory' [Called from Wikimedia\Services\ServiceContainer::{closure} in /var/www/wiki/includes/ServiceWiring.php at line 383] in /var/www/wiki/includes/debug/MWDebug.php on line 376
PHP Deprecated:  Premature access to service 'MainWANObjectCache' [Called from Wikimedia\Services\ServiceContainer::{closure} in /var/www/wiki/includes/ServiceWiring.php at line 406] in /var/www/wiki/includes/debug/MWDebug.php on line 376
PHP Deprecated:  Premature access to service 'LocalServerObjectCache' [Called from Wikimedia\Services\ServiceContainer::{closure} in /var/www/wiki/includes/ServiceWiring.php at line 411] in /var/www/wiki/includes/debug/MWDebug.php on line 376
PHP Deprecated:  Premature access to service 'ConfiguredReadOnlyMode' [Called from Wikimedia\Services\ServiceContainer::{closure} in /var/www/wiki/includes/ServiceWiring.php at line 420] in /var/www/wiki/includes/debug/MWDebug.php on line 376
```
`php /var/www/wiki/maintenance/update.php`:

```
PHP Deprecated:  Premature access to service container [Called from Hooks::run in /var/www/wiki/includes/Hooks.php at line 135] in /var/www/wiki/includes/debug/MWDebug.php on line 376
PHP Deprecated:  Premature access to service 'HookContainer' [Called from MediaWiki\MediaWikiServices::getInstance in /var/www/wiki/includes/MediaWikiServices.php at line 252] in /var/www/wiki/includes/debug/MWDebug.php on line 376
PHP Deprecated:  Premature access to service 'ObjectFactory' [Called from Wikimedia\Services\ServiceContainer::{closure} in /var/www/wiki/includes/ServiceWiring.php at line 535] in /var/www/wiki/includes/debug/MWDebug.php on line 376
PHP Deprecated:  Premature access to service 'StatsdDataFactory' [Called from ObjectCache::newFromParams in /var/www/wiki/includes/objectcache/ObjectCache.php at line 149] in /var/www/wiki/includes/debug/MWDebug.php on line 376
PHP Deprecated:  Premature access to service 'MainConfig' [Called from Wikimedia\Services\ServiceContainer::{closure} in /var/www/wiki/includes/ServiceWiring.php at line 1424] in /var/www/wiki/includes/debug/MWDebug.php on line 376
PHP Deprecated:  Premature access to service 'ConfigFactory' [Called from Wikimedia\Services\ServiceContainer::{closure} in /var/www/wiki/includes/ServiceWiring.php at line 734] in /var/www/wiki/includes/debug/MWDebug.php on line 376
PHP Deprecated:  Premature access to service 'BootstrapConfig' [Called from Wikimedia\Services\ServiceContainer::{closure} in /var/www/wiki/includes/ServiceWiring.php at line 277] in /var/www/wiki/includes/debug/MWDebug.php on line 376
PHP Deprecated:  Premature access to service 'DBLoadBalancer' [Called from ObjectCache::{closure} in /var/www/wiki/includes/objectcache/ObjectCache.php at line 181] in /var/www/wiki/includes/debug/MWDebug.php on line 376
PHP Deprecated:  Premature access to service 'DBLoadBalancerFactory' [Called from Wikimedia\Services\ServiceContainer::{closure} in /var/www/wiki/includes/ServiceWiring.php at line 383] in /var/www/wiki/includes/debug/MWDebug.php on line 376
PHP Deprecated:  Premature access to service 'MainWANObjectCache' [Called from Wikimedia\Services\ServiceContainer::{closure} in /var/www/wiki/includes/ServiceWiring.php at line 406] in /var/www/wiki/includes/debug/MWDebug.php on line 376
PHP Deprecated:  Premature access to service 'LocalServerObjectCache' [Called from Wikimedia\Services\ServiceContainer::{closure} in /var/www/wiki/includes/ServiceWiring.php at line 411] in /var/www/wiki/includes/debug/MWDebug.php on line 376
PHP Deprecated:  Premature access to service 'ConfiguredReadOnlyMode' [Called from Wikimedia\Services\ServiceContainer::{closure} in /var/www/wiki/includes/ServiceWiring.php at line 420] in /var/www/wiki/includes/debug/MWDebug.php on line 376
MediaWiki 1.36.0 Updater

Your composer.lock file is up to date with current dependencies!
Going to run database updates for wiki
Depending on the size of your database this may take a while!
Abort with control-c in the next five seconds (skip this countdown with --quick) ...
[...]
```

**Comment by samrose** [comment]
Still seeing this same issue now

**Comment by Seb35** [comment]
Officially, there is indeed no support for MW 1.36, but in fact it works quite well (and also with 1.37-alpha). There are quite a lot of deprecations since MW 1.35 (and it continues with 1.36 and 1.37), so there is a lot of work to convert all these deprecation warnings but work is ongoing.

**Comment by Abijeet** [comment]
This was fixed in #5021
